### ISSUE TITLE
use `assign` instead of `merge` here ?

### ISSUE THREAD

**Issue by bartocc** [emberjs/data#5817]
I might be wrong, but shouldn't we use `Ember.assign` instead of `Ember.merge` here to avoid a 3.6 deprecation warning ?

https://github.com/emberjs/data/blob/4c539f466a21e91499ff18ba0318827db1f27252/addon/-private/system/store/finders.js#L11

**Comment by mhankus** [comment]
Deprecation warnings are also caused by (in 3.7)
[data/addon/-private/system/model/errors.js](https://github.com/emberjs/data/blob/v3.7.0/addon/-private/system/model/errors.js)
```
errorsFor(attribute) {
    let map = get(this, 'errorsByAttributeName');

    if (!map.has(attribute)) {
      map.set(attribute, new A());
    }

    return map.get(attribute);
  },
```
https://deprecations-app-prod.herokuapp.com/deprecations/v3.x/#toc_array-new-array-wrapper

**Comment by rwjblue** [comment]
Looks like https://github.com/emberjs/data/pull/5812/files is already open for fixing the `Ember.merge` deprecation.

**Comment by igorT** [comment]
This was fixed by #5812, thx @BryanCrotaz
